noxfile.py

```diff
@@ -0,0 +1,34 @@
+import nox
+from nox_poetry import session
+
+
+PYTHON_VERSIONS = ["3.6", "3.7", "3.8", "3.9"]
+SPHINX_VERSIONS = ["2.2", "2.3", "2.4", "3.0", "3.2"]
+TEST_DEPENDENCIES = [
+    "nose",
+    "sphinx_testing",
+    "sphinxcontrib.plantuml",
+    "matplotlib",
+    "responses",
+    "sphinx_copybutton",
+]
+
+
+def is_supported(python: str, sphinx: str) -> bool:
+    return not (python == "3.6" and float(sphinx) > 3.0)
+
+
+def run_tests(session, sphinx):
+    session.install(".")
+    session.install(*TEST_DEPENDENCIES)
+    session.run("pip", "install", f"sphinx=={sphinx}", silent=True)
+    session.run("make", "test", external=True)
```

Comment:
ok sure, that's actually how i originally had it. Happy to change it back.

I'll outline my reasons for calling the makefile

1. developers don't need to use nox, or install the additional nox system dependencies to work on and test this project. Nox is *only* required if you wish to run the full matrix test suite
2. the makefile is both a convenience, and an entrypoint for developers to see the list of available commands. As you say, it also provides an abstraction. (you only need to know you want to 'test', for example, you don't need to remember the full command
3. i elected to call the makefile from the noxfile so that there was a 'single source of truth' for the commands. otherwise the exact same logic is duplicated between the makefile and the noxfile. If you ever change one, you have to remember to change the other.

The Makefile becomes more handy if you add additional targets, for example
- to lint, the command is probably `poetry run flake8 sphinxcontrib/ tests/ noxfile.py` (compare that to `make lint`)
- to format, the command might be something like `poetry run black sphinxcontrib/ tests/ noxfile.py && poetry run isort sphinxcontrib/ tests/ noxfile.py` (compared to `make format`)

you can see why the Makefile becomes a convenience...

Reply:
I see. 
For this use case I just used tox :)
``tox -e flake8``, ``tox -e docs``, ``tox`` for everything.
Benefit was that tox created a clean python env.

But this benefit we already get with poetry now.
So creating another env for tests inside a "poetry env" makes no sense and costs time.

So after your arguments I like the "make" idea and support it.
The ``Makefile``will be the place to put all logic for running tests, linter and co.
``nox`` will make use of it, as nox/tox is no longer the default option to call stuff for the developer.